<a href="https://colab.research.google.com/github/isilberfin/Data-Augmentation-NLP/blob/main/sim_test.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# CODE

In [ ]:
#importing csv files having the columns of all predictions for augmented test sets

In [ ]:
import pandas as pd
amazon = pd.read_csv("comparison_all.csv")

In [ ]:
amazon.head(3)

Y  y_pred_all  y_pred_all_ins  y_pred_all_syn  y_pred_all_swap  \
0  1           1               1               1                1   
1  1           1               1               1                1   
2  1           1               1               1                1   

   y_pred_all_swap2  y_pred_all_exp2  
0                 1                1  
1                 1                1  
2                 1                1

In [ ]:
# the number matching columns will be detected df by df

In [ ]:
matching = (amazon.y_pred_all.eq(amazon.y_pred_all_ins)
                       .value_counts()
                       .rename({True:'mach', False: 'no match'})
                       .rename_axis('state')
                       .reset_index(name='count'))

In [ ]:
matching

state  count
0      mach    491
1  no match      9

In [ ]:
#df = pd.DataFrame({'y_pred_ins': [476],'y_pred_syn': [454],'y_pred_swap': [478],'y_pred_swap2': [470],'y_pred_exp2': [491]},index = ["y_pred"])

#ALL-DATA

In [ ]:
df

y_pred  y_pred_ins  y_pred_syn  y_pred_swap  y_pred_swap2  y_pred_exp2
y_test     407         409         399          407           413          410

In [ ]:
df

y_pred_ins  y_pred_syn  y_pred_swap  y_pred_swap2  y_pred_exp2
y_pred         476         454          478           470          491

#### eşleşmeler (all)
* y_test ve y_pred arasında 407 benzeşme 93 farklı
* y_test ve y_pred_all_ins arasında 409 benzeşme 91 farklı
* y_test ve y_pred_all_syn arasında 399 benzeşme 101 farklı
* y_test ve y_pred_all_swap arasında 407 benzeşme 93 farklı
* y_test ve y_pred_all_swap2 arasında 413 benzeşme 87 farklı
* y_test ve y_pred_all_exp2 arasında 410 benzeşme 90 fark
********
y_pred_alla benzeme açısından:   
* y_pred_all_ins : 476 benzeşme 24 farklı
* y_pred_all_syn : 454 benzeşme 46 farklı
* y_pred_all_swap : 478 benzeşme 22 farklı
* y_pred_all_swap2 : 470 benzeşme 30 farklı
* y_pred_all_exp2 : 491 benzeşme 9 farklı

#TWITTER

In [ ]:
df

y_pred  y_pred_ins  y_pred_syn  y_pred_swap  y_pred_swap2  y_pred_exp2
y_test     718         710         715          710           709          718

In [ ]:
df

y_pred_ins  y_pred_syn  y_pred_swap  y_pred_swap2  y_pred_exp2
y_pred         781         770          779           780          789

#### eşleşmeler (twitter)
* y_test ve y_pred arasında eşleşen 718 eşleşmeyen 73 etiket
* y_test ve y_pred_tw_insert arasında eşleşen 710 eşleşmeyen 81 etiket
* y_test ve y_pred_tw_syn arasında eşleşen 715 eşleşmeyen 76 etiket
* y_test ve y_pred_tw_swap arasında eşleşen 710 eşleşmeyen 81 etiket
* y_test ve y_pred_tw_swap2 arasında eşleşen 709 eşleşmeyen 82 etiket
* y_test ve y_test_tw_exp2 arasında eşleşen 718 eşleşmeyen 73 etiket
************
y_pred_twittera benzeme açısından:
* y_pred_tw_insert : 781 benzeşme 10 farklı
* y_pred_tw_syn : 770 eşleşme 21 farklı
* y_pred_tw_swap : 779 benzeşme 12 farklı
* y_pred_tw_swap2 : 780 benzeşme 11 farklı
* y_pred_tw_exp2 : 789 benzeşme 2 farklı

# AMAZON

In [ ]:
df

y_pred  y_pred_ins  y_pred_syn  y_pred_swap  y_pred_swap2  y_pred_exp2
y_test     178         173         168          174           175          175

In [ ]:
df

y_pred_ins  y_pred_syn  y_pred_swap  y_pred_swap2  y_pred_exp2
y_pred         191         184          192           193          195

#### eşleşmeler (amazon)
* amazon y_test ve y_pred arasında eşleşen 178 etiket eşleşmeyen 22 etiket
* amazon y_test ve y_pred_amazon_ins arasında eşleşen 173 etiket ve eşleşmeyen 27 etiket
* amazon y_test ve y_pred_amzon_syn arasında eşleşen 168 etiket eşleşmeyen 32 etiket
* amazon y_test ve y_pred_amazon_swap arasında eşleşen 174 etiket eşleşmeyen 26 etiket
* amazon y_test ve y_pred_amazon_swap2 arasında eşleşen 175 etiket eşleşmeyen 25 etiket
* amazon y_test ve y_pred_amazon_exp2 arasında eşleşen 175 etiket eşleşmeyen 25 etiket

*******
y_pred_amazona benzeme açısından :
* y_pred_amazon_ins : 191 eşleşme 9 farklı
* y_pred_amazon_syn : 184 eşleşme 16 farklı
* y_pred_amazon_swap : 192 eşleşme 8 farklı
* y_pred_amazon_swap2 : 193 eşleşme 7 farklı
* y_pred_amazon_exp2 : 195 eşleşme 5 farklı

*******
